### PR TITLE
Extended the plugin

### PR DESCRIPTION
**Pull request by MrDoe**
* Added .slsql support for datasources in SQL
* Fixed comment blocks for SSL datasources
* Fixed SQL highlighting in strings
* Added navigator variables highlighting
* Removed separate "IIf" and added it to blue functions

I guess we can shorten the sql.tmLanguage.json file and just use source.sql. I'm working on it...

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,13 +22,28 @@
                     ".ssl",".srvscr"
                 ],
                 "configuration": "./language-configuration.json"
+            },
+            {
+                "id": "SQL",
```
Comment (jbouecke) [inline package.json:27]: Since it's a very specific format, we should not use the generic ID: SQL. How about slsql or so?

File: package.json [hunk 2]
```diff
@@ -22,13 +22,28 @@
...
         ],
         "grammars": [
             {
                 "language": "SSL",
                 "scopeName": "source.ssl",
                 "path": "./syntaxes/ssl.tmLanguage.json"
+            },
+            {
+                "language": "SQL",
+                "scopeName": "source.sql",
```
Comment (jbouecke) [inline package.json:45]: see above: I suggest 'source.slsql'

File: language-configuration.json [hunk 3]
```diff
@@ -1,10 +1,8 @@
 {
   "comments": {
-    // symbols used for start and end a block comment. Remove this entry if your language does not support block comments
-    "blockComment": ["/*", ";"],
+    "blockComment": ["/*", "*/|;"],
```
Comment (jbouecke) [inline language-configuration.json:3]: This is wrong. A comment ends always only with a ';'.

File: syntaxes/ssl.tmLanguage.json [hunk 4]
```diff
@@ -130,48 +139,14 @@
...
-          "end": "(?=,)|(?=\\))",
-          "patterns": [
-            {
-              "include": "#expression"
-            }
-          ]
-        }
-      ]
-    },
-    "iif_in_sql": {
```
Comment (jbouecke) [inline syntaxes/ssl.tmLanguage.json:152]: The special teratment off "iif" was done by purpose to support a common pattern of using iif within sql statements. See the following higlightings (first is with my original version, second with yours).

image

image

File: syntaxes/ssl.tmLanguage.json [hunk 5]
```diff
@@ -72,13 +75,19 @@
...
+        {
+          "include": "#variableInSQLString"
         }
       ]
     },
     "comment": {
       "name": "comment.block.srvsrc",
       "begin": "/\\*",
-      "end": ";"
+      "end": "\\*/|;"
```
Comment (jbouecke) [inline syntaxes/ssl.tmLanguage.json:90]: This is wrong! The comment ends always only with a ";".

Comment (MrDoe) [reply syntaxes/ssl.tmLanguage.json:90]: Sorry, you are right! I confused it with datasources written in SQL.

File: language-configuration.json (hunk 3 above)
Comment (MrDoe) [reply language-configuration.json:3]: Sorry, you are right! I confused it with datasources written in SQL.

File: syntaxes/ssl.tmLanguage.json [hunk 6]
```diff
@@ -188,8 +163,8 @@
       ]
     },
     "langConstants": {
-      "name": "constant.language",
-      "match": "\\.(t|T|f|F)\\."
+      "name": "constant.character.escape",
```
Comment (jbouecke) [inline syntaxes/ssl.tmLanguage.json:166]: This is not a character escape but a language constant.

File: syntaxes/ssl.tmLanguage.json [hunk 7]
```diff
@@ -209,92 +184,40 @@
       ]
     },
     "stringsDouble": {
-      "name": "string.quoted.double",
+      "name": "meta.preprocessor.string",
```
Comment (jbouecke) [inline syntaxes/ssl.tmLanguage.json:187]: It's not a meta preprocessor sting, but just a standard string

File: syntaxes/ssl.tmLanguage.json [hunk 8]
```diff
@@ -209,92 +184,40 @@
...
       "begin": "\"",
       "end": "\""
     },
     "stringsSingle": {
-      "name": "string.quoted.sinlge",
+      "name": "meta.preprocessor.string",
```
Comment (jbouecke) [inline syntaxes/ssl.tmLanguage.json:192]: see above

File: syntaxes/ssl.tmLanguage.json (hunk 6 above)
Comment (MrDoe) [reply syntaxes/ssl.tmLanguage.json:166]: I know, I used the name to change the color to orange, if the vscode modern-dark-theme is being used. Is there a way to define a color independently from the used theme? If not, I propose we create a custom vscode theme for SSL.

File: syntaxes/ssl.tmLanguage.json (hunk 7 above)
Comment (MrDoe) [reply syntaxes/ssl.tmLanguage.json:187]: I used this name just for changing the color.

Comment (MrDoe) [reply syntaxes/ssl.tmLanguage.json:187]: see above

File: syntaxes/ssl.tmLanguage.json (hunk 8 above)
Comment (MrDoe) [reply syntaxes/ssl.tmLanguage.json:192]: see above

File: syntaxes/ssl.tmLanguage.json (hunk 6 above)
Comment (jbouecke) [reply syntaxes/ssl.tmLanguage.json:166]: I think it's good to stick with the more precise semantic definition and just modify the scheme to your preference. This is what themes are about.
And I like the idea of creating an SSL theme that e.g. makes blue function blue again ;-)

**Pull request closed without merging**